Add verbiage about feature owners (#2977)
sourcegraph/handbook: content/departments/product-engineering/engineering/code-graph/code-insights/processes.md

  - We hold a [fortnightly synchronous iteration planning](#fortnightly-iteration-planning), we take a look at the proposed issues together on the ["Next iteration" view](https://github.com/orgs/sourcegraph/projects/200/views/4).

> - Larger issues that involve multiple domains may benefit from a "feature owner". A feature owner assumes full responsibility for gathering all requirements to complete an issue. The feature owner is expected to have full understanding of the issue and pull in any resources or team members needed to complete the task.

Reviewer: @Joelkw's PR called this "project lead", should probably align the wording with that so it's consistent within the same handbook page (clear that it refers to the same person).

- If one or more issues that were planned for an iteration are looking to **not get finished** (which includes testing) in the [current iteration](https://github.com/orgs/sourcegraph/projects/200/views/1) (while maintaining sustainable work practices) the assignee **raises this as soon as possible asynchronously** to the team (including the PM and EM), e.g. on the GitHub issue or Slack. These issues then **become _proposed_ issues for the next iteration** (meaning nothing carries over automatically, but we also don't just drop and forget missed issues).

## Project Leads

> Larger issues that involve multiple domains may benefit from a "project lead". A project lead assumes full responsibility for gathering all requirements to complete an issue. The project lead is expected to have full understanding of the issue and pull in any resources or team members needed to complete the task.

Reviewer 2: Do we want to call these "issues" or "projects"? (Confusion with GH issues, or is this intentional/there will be per project issues as well?) (Update: I see we're using tracking issues, so this point is moot!) 

Also, do they necessarily have to involve multiple domains? (Monorepo improvements feels like primarily a backend project, for example?) 

```suggestion
Larger issues, especially ones that involve multiple domains, may benefit from a "project lead". A project lead assumes full responsibility for gathering all requirements to complete an issue. The project lead is expected to have full understanding of the issue and pull in any resources or team members needed to complete the task.
```

### Time commitment

Being a project lead takes a bit more time than just working on a single issue. If you are a project lead then your capacity will be reduced. Project leads should make sure to take this into consideration when planning other iteration work.

### Responsibilities

> 1. Have a full understanding of the requirements and the work that needs to be done across the frontend and backend. The main benefit here is catching any misunderstandings or integration issues before they happen.

Reviewer 2: ```suggestion
1. Be the first point of engineering contact for product and design
1. Have a full understanding of the requirements and the work that needs to be done across the frontend and backend. The main benefit here is catching any misunderstandings or integration issues before they happen.
```

- If one or more issues that were planned for an iteration are looking to **not get finished** (which includes testing) in the [current iteration](https://github.com/orgs/sourcegraph/projects/200/views/1) (while maintaining sustainable work practices) the assignee **raises this as soon as possible asynchronously** to the team (including the PM and EM), e.g. on the GitHub issue or Slack. These issues then **become _proposed_ issues for the next iteration** (meaning nothing carries over automatically, but we also don't just drop and forget missed issues).

## Project Leads

> Larger issues, especially ones that involve multiple domains, may benefit from a "project lead". A project lead assumes full responsibility for gathering all requirements to complete an issue. The project lead is expected to have full understanding of the issue and pull in any resources or team members needed to complete the task.

Author: ```suggestion
Larger projects, especially ones that involve multiple domains, may benefit from a "project lead". A project lead assumes full responsibility for gathering all requirements to complete a project. The project lead is expected to have full understanding of the issue and pull in any resources or team members needed to complete the task.
```
